stylix/target.nix

```
@@ -38,14 +38,36 @@
     {
       mkEnableTarget =
         humanName: autoEnable:
+        config.lib.stylix.mkEnableTargetWith { inherit humanName autoEnable; };
+
+      mkEnableTargetWith =
+        {
+          humanName,
+          autoEnable ? true,
+          autoEnableExpr ? null,
+          defaultText ? null,
+          example ?
+            if args ? autoEnableExpr || args ? defaultTex then true else !autoEnable,
+        }@args:
+        assert !(args ? autoEnableExpr && args ? defaultText);
+        let
+          wrapExpr = expr: if lib.strings.hasInfix " " expr then "(${expr})" else expr;
+        in
         lib.mkEnableOption "theming for ${humanName}"
         // {
           default = cfg.enable && cfg.autoEnable && autoEnable;
-          example = !autoEnable;
-        }
-        // lib.optionalAttrs autoEnable {
-          defaultText = lib.literalMD "same as `stylix.autoEnable`";
+          defaultText =
+            if args ? defaultText then
+              defaultText
+            else if args ? autoEnableExpr then
+              lib.literalExpression "stylix.enable && stylix.autoEnable && ${wrapExpr autoEnableExpr}"
```

Thinking about ideas for using the old markdown-based default text :thinking: 

```suggestion
              lib.literalMD "same as `stylix.autoEnable`, ${autoEnableMD}"
```

or

```suggestion
              lib.literalMD "same as `stylix.autoEnable`, when `` ${autoEnableExpr} ``"
```

or ...?

---

I think half the issue here is that the default is actually incorrect. I've opened #1255 to fix it.

With that merged, this can be simplified to

```suggestion
              lib.literalExpression "stylix.autoEnable && ${wrapExpr autoEnableExpr}"
```